test/Microsoft.AspNet.Mvc.TagHelpers.Test/ScriptTagHelperTest.cs

```diff
@@ -580,17 +595,18 @@ public static TheoryData DoesNotRunWhenARequiredAttributeIsMissing_Data
             var globbingUrlBuilder = new Mock<GlobbingUrlBuilder>();
             globbingUrlBuilder.Setup(g => g.BuildUrlList("/js/site.js", "**/*.js", null))
                 .Returns(new[] { "/js/site.js", "/common.js" });
-            var helper = new ScriptTagHelper
+
+            var helper = new ScriptTagHelper(
+                logger.Object,
+                hostingEnvironment,
+                MakeCache(),
+                new CommonTestEncoder(),
+                new CommonTestEncoder())
             {
                 GlobbingUrlBuilder = globbingUrlBuilder.Object,
-                Logger = logger.Object,
-                HostingEnvironment = hostingEnvironment,
                 ViewContext = viewContext,
                 Src = "/js/site.js",
                 SrcInclude = "**/*.js",
-                HtmlEncoder = new CommonTestEncoder(),
-                JavaScriptEncoder = new TestJavaScriptEncoder(),
```

Comment:
Stick with `TestJavaScriptEncoder` where it was used before. Suspect that was intentional but, even if not, your changes really should not change assertions at all in these unit tests.